src/components/Heading/Heading.tsx

```diff
@@ -25,52 +28,69 @@ export type HeadingTypes =
 
 export type HeadingTagTypes = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'span' | 'legend'
 
-type ElementProps = Omit<HTMLAttributes<HTMLElement>, keyof Props | keyof TextProps>
+type ElementProps = Omit<
+  HTMLAttributes<HTMLElement>,
+  keyof Props | keyof TextProps | 'role' | 'aria-level'
+>
+
+const generateTagProps = (tag: HeadingTagTypes | undefined, level: number) => {
```

Comment:
IMO: 意図的でなければ undefined は慣習? に倣っておきたい。

```suggestion
const generateTagProps = (tag?: HeadingTagTypes, level: number) => {
```

Reply:
この関数はコンポーネント内部からしか呼ばれないため、引数が存在しないわけではなく、明確にundefinedが渡されるので意図的にundefinedを型指定しています〜

Reply:
そういう意味だと、以下のように `level` を前にして `tag?` をオプショナルとして扱うことで、基本は `level` の数値をもとに tag を生成することがわかりやすくなりつつ、明示的に `tag` を指定した場合はそれに基づいて処理が行われることがわかりやすいかなと思いました

```TypeScript
const generateTagProps = (level: number, tag?: HeadingTagTypes, ) => {
```

これもIMOなのでそこまで強い意志はないですが

Reply:
@uknmr @yamish123 
私も強い意志はない & yamishさんの案が良さそうだったので対応しました
https://github.com/kufu/smarthr-ui/pull/3489/commits/6099c48659756258fa419577d986ec5a356c0a97

Reply:
良さそう！